xbmc/cores/RetroPlayer/rendering/VideoRenderers/RPRendererOpenGLES.cpp

```diff
@@ -296,14 +342,63 @@ void CRPRendererOpenGLES::Render(uint8_t alpha)
   rect.y1 /= m_sourceHeight;
   rect.y2 /= m_sourceHeight;
 
-  glTexCoord2f(rect.x1, rect.y1);  glVertex2f(m_rotatedDestCoords[0].x, m_rotatedDestCoords[0].y);
-  glTexCoord2f(rect.x2, rect.y1);  glVertex2f(m_rotatedDestCoords[1].x, m_rotatedDestCoords[1].y);
-  glTexCoord2f(rect.x2, rect.y2);  glVertex2f(m_rotatedDestCoords[2].x, m_rotatedDestCoords[2].y);
-  glTexCoord2f(rect.x1, rect.y2);  glVertex2f(m_rotatedDestCoords[3].x, m_rotatedDestCoords[3].y);
+  float u1 = rect.x1;
+  float u2 = rect.x2;
+  float v1 = rect.y1;
+  float v2 = rect.y2;
+
+  const uint32_t color = (alpha << 24) | 0xFFFFFF;
+
+  glBindTexture(m_textureTarget, renderBuffer->TextureID());
+
+  GLint filter = (GetRenderSettings().VideoSettings().GetScalingMethod() == VS_SCALINGMETHOD_NEAREST ? GL_NEAREST : GL_LINEAR);
```

Comment:
Ternary operator makes the line a little long, which is hard to read on small screens. How about

```c++
GLint filter = GL_NEAREST;
if (GetRenderSettings().VideoSettings().GetScalingMethod() == VS_SCALINGMETHOD_LINEAR)
  filter = GL_LINEAR;
```